bokeh/core/property/nullable.py

```diff
@@ -73,12 +73,14 @@ def validate(self, value: Any, detail: bool = True) -> None:
         msg = "" if not detail else f"expected either None or a value of type {self.type_param}, got {value!r}"
         raise ValueError(msg)
 
-class NonNullable(SingleParameterizedProperty[T]):
-    """ A property accepting a value of some other type while having undefined default. """
+class NonNullable(Required[T]):
+    """
+    A property accepting a value of some other type while having undefined default.
 
-    def __init__(self, type_param: TypeOrInst[Property[T]], *, default: Init[T] = Undefined,
-            help: str | None = None, serialized: bool | None = None, readonly: bool = False) -> None:
-        super().__init__(type_param, default=default, help=help, serialized=serialized, readonly=readonly)
```

Comment:
Why not keep an initializer that calls `deprecated`? Then the fact of the deprecation can be maintained under test as well.